lib/sisimai/bite/email/ezweb.rb

```diff
@@ -0,0 +1,260 @@
+          # Your message was not delivered within 0 days and 1 hours.
+          # Remote host is not responding.
+          %r/Your message was not delivered within /,
+        ],
+        onhold: [
+          %r/Each of the following recipients was rejected by a remote mail server/,
+        ],
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'au EZweb: http://www.au.kddi.com/mobile/'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-SPASIGN']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from au EZweb
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['from']    =~ Re0[:from]
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['received'].find { |a| a =~ Re0[:received] }
+        if mhead['message-id']
+          match += 1 if mhead['message-id'] =~ Re0[:'message-id']
+        end
+        return nil if match < 2
+
+        require 'sisimai/string'
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        rxboundary = %r/\A__SISIMAI_PSEUDO_BOUNDARY__\z/
+        v = nil
+
+        if mhead['content-type']
+          # Get the boundary string and set regular expression for matching with
+          # the boundary string.
+          require 'sisimai/mime'
+          b0 = Sisimai::MIME.boundary(mhead['content-type'], 1)
+          if b0.size > 0
+            # Convert to regular expression
+            rxboundary = Regexp.new('\A' + Regexp.escape(b0) + '\z')
+          end
+        end
+        rxmessages = []
+        ReFailure.each_key { |a| rxmessages.concat(ReFailure[a]) }
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
```

Comment:
__[rubocop]__
* Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. [:ref](https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)

<!-- Comment by [SideCI](https://sideci.com) -->